main.go

```diff
@@ -0,0 +1,517 @@
+	}
+	config.Import(path)
+	prog, err := config.Load()
+	if err != nil {
+		panic(err)
+	}
+
+	for _, pkgInfo := range prog.InitialPackages() {
+		(*pkgs)[path] = Package{
+			Fset:      prog.Fset,
+			Files:     pkgInfo.Files,
+			Path:      pkgInfo.Pkg.Path(),
+			TypesInfo: &pkgInfo.Info,
+		}
+	}
+
+	if p, ok := (*pkgs)[path]; ok {
+		return &p
+	}
+
+	return nil
+}
+
+func WriteIndent(context *Context) {
+	for i := 0; i < context.Indent; i++ {
+		fmt.Printf("  ")
+	}
+}
+
+func WriteFieldStruct(context *Context, f *ast.Field) {
+	t := Field_GetType(context.Package, f)
+
+	n := Type_GetNamed(t)
+	if n == nil {
+		return
+	}
+
+	x := Type_LoadPackage(&context.Packages, n)
+	if x == nil {
+		return
+	}
+
+	ProcessStruct(context.StartStruct(x), n.Obj().Name())
+}
+
+func WriteComment(c *Context, g *ast.CommentGroup) {
+	// comment regex
+	rc := regexp.MustCompile("^\\/\\/(.*)")
+	// "uncommented" regex
+	uc := regexp.MustCompile("^\\s*!td:unc\\s?(.*)")
+	// "skip" regex
+	sr := regexp.MustCompile("^\\s*!td:skip\\s*$")
+	// "follow" regex
+	fr := regexp.MustCompile("^\\s*!td:follow\\s*$")
+
+	has_output := false
+	for _, l := range g.List {
+		// The current approach supports single-line comments. There is
+		// ambiguity regarding line prefixes and whitespace when parsing
+		// block comments. Since single-line comments are consistent, it
+		// avoids this ambiguity.
+		m := rc.FindStringSubmatch(l.Text)
+		if m == nil {
+			continue
+		}
+
+		// don't write flag
+		if f := sr.FindStringSubmatch(m[1]); f != nil {
+			continue
+		}
+		// don't write flag
+		if f := fr.FindStringSubmatch(m[1]); f != nil {
+			continue
+		}
+
+		// Do not emit a "#" when the line is intended to be uncommented.
+		// This is useful for supplying default values.
+		ug := uc.FindStringSubmatch(m[1])
+		if ug != nil {
+			has_output = true
+			c.WriteLn(ug[1])
+		} else {
+			for i, _ := range m[1] {
+				if i+1 >= len(m[1]) ||
+					m[1][i+0] != '/' ||
+					m[1][i+1] != '/' {
+					break
+				}
+				// Rewrite the line such that additional "//" is converted
+				// into "#". If any other character (including whitespace)
+				// is encountered, stop rewriting the string. This makes
+				// it possible to have section headings.
+				m[1] = m[1][:i] + "#" + m[1][i+2:]
+			}
+			has_output = true
+			c.WriteLn("#" + m[1])
+		}
+	}
+
+	if has_output {
+		c.WriteLn("")
+	}
+}
+
+const (
+	TD_NONE = 0
+	// The default behavior is to write the documentation for all
+	// exported fields. Sometimes this behavior is undesirable.
+	// Enabling this flag with "!td:skip" skips writing
+	// documentation for the current field.
+	TD_SKIP = 1
+	// Structures are not automatically followed. This prevents generating
+	// documentation for types from external packages. This "opt-in"
+	// behavior also prevents infinite recursion.
+	TD_FOLLOW = 2
+)
+
+func ParseCommentFlags(g *ast.CommentGroup) int {
+	// comment regex
+	cr := regexp.MustCompile("^\\/\\/(.*)")
+	// "skip" regex
```

Comment:
As above, if performance is a concern, could be moved to the package level for single evaluation.